tests/ops/op_math/test_prod.py

```diff
@@ -0,0 +1,781 @@
+        true_mat = jnp.array(true_mat)
+
+        assert jnp.allclose(mat, true_mat)
+
+    @pytest.mark.torch
+    def test_prod_torch(self):
+        """Test matrix is cast correctly using torch parameters."""
+        import torch
+
+        theta = torch.tensor(1.23)
+        rot_params = torch.tensor([0.12, 3.45, 6.78])
+
+        prod_op = Prod(
+            qml.Rot(rot_params[0], rot_params[1], rot_params[2], wires=0),
+            qml.RX(theta, wires=1),
+            qml.Identity(wires=0),
+        )
+        mat = prod_op.matrix()
+
+        true_mat = (
+            qnp.kron(gd.Rot3(rot_params[0], rot_params[1], rot_params[2]), qnp.eye(2))
+            @ qnp.kron(qnp.eye(2), gd.Rotx(theta))
+            @ qnp.eye(4)
+        )
+        true_mat = torch.tensor(true_mat)
+        true_mat = torch.tensor(true_mat, dtype=torch.complex64)
+
+        assert torch.allclose(mat, true_mat)
+
+    @pytest.mark.tf
+    def test_prod_tf(self):
+        """Test matrix is cast correctly using tf parameters."""
+        import tensorflow as tf
+
+        theta = tf.Variable(1.23)
+        rot_params = tf.Variable([0.12, 3.45, 6.78])
+
+        prod_op = Prod(
+            qml.Rot(rot_params[0], rot_params[1], rot_params[2], wires=0),
+            qml.RX(theta, wires=1),
+            qml.Identity(wires=0),
+        )
+        mat = prod_op.matrix()
+
+        true_mat = (
+            qnp.kron(gd.Rot3(0.12, 3.45, 6.78), qnp.eye(2))
+            @ qnp.kron(qnp.eye(2), gd.Rotx(1.23))
+            @ qnp.eye(4)
+        )
+        true_mat = tf.Variable(true_mat)
+        true_mat = tf.Variable(true_mat, dtype=tf.complex128)
+
+        assert isinstance(mat, tf.Tensor)
+        assert mat.dtype == true_mat.dtype
+        assert np.allclose(mat, true_mat)
+
+
+class TestProperties:
+    """Test class properties."""
+
+    @pytest.mark.parametrize("ops_lst", ops)
+    def test_num_params(self, ops_lst):
+        """Test num_params property updates correctly."""
+        prod_op = Prod(*ops_lst)
+        true_num_params = sum(op.num_params for op in ops_lst)
+
+        assert prod_op.num_params == true_num_params
+
+    @pytest.mark.parametrize("ops_lst", ops)
+    def test_num_wires(self, ops_lst):
+        """Test num_wires property updates correctly."""
+        prod_op = Prod(*ops_lst)
+        true_wires = set()
+
+        for op in ops_lst:
+            true_wires = true_wires.union(op.wires.toset())
+
+        assert prod_op.num_wires == len(true_wires)
+
+    @pytest.mark.parametrize(
+        "ops_lst, hermitian_status",
+        [(ops_tup, status) for ops_tup, status in zip(ops, ops_hermitian_status)],
+    )
+    def test_is_hermitian(self, ops_lst, hermitian_status):
+        """Test is_hermitian property updates correctly."""
+        prod_op = prod(*ops_lst)
+        assert prod_op.is_hermitian == hermitian_status
+
+    @pytest.mark.tf
+    @pytest.mark.xfail  # this will fail until we can support is_hermitian checks for parametric ops
```

Comment:
All the interface is_hermitian tests have been marked with xfail because the new simple check for hermiticity relies on each operations `is_hermitian` property, which is by default False for any parametric operation. This prevents parametric operations from being compatible with the new quick check.